apps/demo/app/entry.client.tsx

```diff
@@ -1,4 +1,10 @@
 import { RemixBrowser } from "@remix-run/react";
-// import { hydrate } from "react-dom";
+import { hydrateRoot } from "react-dom/client";
+import { StrictMode } from "react";
 
-// hydrate(<RemixBrowser />, document);
+hydrateRoot(
```

Comment:
hydrate is deprecated, upgrade to use `hydrateRoot` instead. The `strictmode` gave me a better error message of the root cause for the infinitive loop.